### PR TITLE
chore(release): guard develop ruleset review enforcement for soundness paths

### PR DESCRIPTION
**Pull request by joeharris76**
Follow-up to `auto-merge-review-gate-soundness-paths` (#912). The auto-merge **code** side is sound — `make pr-open` and `auto-merge-on-open.yml` share `auto_merge_soundness_paths.py` (`--no-renames`) and withhold auto-merge enablement for soundness-path PRs — but nothing **enforces** a review at the repo layer: the `develop-squash-only` ruleset has `required_approving_review_count: 0` and `require_code_owner_review: false`, so a soundness-path PR can still be squash-merged with zero approvals. This PR delivers w1 + w2; **w3 (the ruleset change) is a deferred repo-admin action** and the item is intentionally **not** marked Completed (TODO stays `In Progress`).

## ⚠️ Deferred admin action required (w3)
The repo-layer gap is closed only when an admin updates the `develop-squash-only` ruleset to `required_approving_review_count >= 1` and `require_code_owner_review: true`. The develop-PR `GITHUB_TOKEN` has no `administration` scope (can't read or write rulesets), so this cannot be done by an agent PR. The exact `gh api` PUT is documented in `docs/operations/repo-admin-settings.md` → "Soundness-path review enforcement (pending admin action)". Until applied, the #912 success_metric stays unmet by design.

## w0 — re-validation (live truth)
- CODEOWNERS assigns `@joeharris76` to all three soundness globs (`benchbox/core/**/validation.py`, `benchbox/core/equivalence/**`, `benchbox/core/query_plans/parsers/**`) — code-owner review is meaningfully ownable.
- Live develop ruleset state (per #912 authoring, unchanged): `required_approving_review_count: 0`, `require_code_owner_review: false`. **Token note:** no `gh`/rulesets-MCP in this env, and the default Actions `GITHUB_TOKEN` lacks `administration` scope — so develop-PR CI cannot read the live ruleset. Per the TODO's explicit fallback, the live check is the documented manual/release-canary step (`RULESET_DRIFT_TOKEN`); the in-CI load-bearing guard is the predicate unit test.

## w1 — predicate + load-bearing unit test
- `_project/scripts/ruleset_review_enforcement.py`: predicate `review_enforcement_findings(rules)` (fails if `required_approving_review_count < 1` OR `require_code_owner_review != true`; accepts both the `rules/branches` list and `rulesets/<id>` object shapes). Narrates the CODEOWNERS soundness globs read from `auto_merge_soundness_paths.SOUNDNESS_PREFIXES` (single source of truth — narration and auto-merge withholding can't drift). CLI for the runbook: `gh api .../rules/branches/develop | … --rules-file -`.
- `tests/unit/release/test_ruleset_review_enforcement.py` (fast lane = required CI): asserts the predicate fails on an unenforced ruleset and passes on an enforced one, plus shape/source-of-truth coverage.
- **Decision recorded:** no live-ruleset develop-PR gate is added — it would either need an admin token the gate doesn't have, or (if forced) go red on the current unenforced ruleset and brick every develop PR while w3 is pending. The live check lives in the documented runbook/canary; the develop-PR meta-check is the predicate logic test.

## w2 — corrected #912 metric + runbook
- Amended `_project/DONE/main/active/auto-merge-review-gate-soundness-paths.yaml` success_metrics: code side complete (withholding); the HARD repo-layer precondition is NOT met by code alone and is enforced by the ruleset change tracked in `auto-merge-review-gate-admin-enforcement`.
- `docs/operations/repo-admin-settings.md`: documents the target state, the manual/canary Verify command (the new predicate), and the admin Apply PUT.

## w4 — load-bearing proof
Reverted the predicate threshold (`count < 1` → `count < 0`) → `test_unenforced_ruleset_fails_on_both_axes` and `test_zero_count_alone_fails` went red. Restored.

## Verification
- `pytest tests/unit/release/ -k ruleset_review_enforcement` → 8 passed (fast lane).
- ruff + ty clean; `make docs-validate` clean; amended #912 YAML validates; TODO graph intact.
- The new `repo-admin-settings.md` subsection uses a `###` heading + separate fenced block, so the existing `scripts/ruleset_drift_check.py` parser still selects the original "Other ruleset properties to preserve" block — verified, its 7 ruleset-drift tests pass.

Independent code review (correctness + honesty): no findings; predicate coalescing edge cases, doc-parser non-breakage, metric honesty, and import resolution all verified. No skipped nits.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01Jb4WFFT3SBDJF57MP4GXYZ)_

### CODE REVIEW

**Comment by joeharris76** [comment]
@codex review

---
_Generated by [Claude Code](https://claude.ai/code)_

File: _project/scripts/ruleset_review_enforcement.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    """
+    params = _pull_request_parameters(rules)
+    if params is None:
+        return [
+            "develop ruleset has no pull_request rule: a soundness-path PR "
+            f"({', '.join(SOUNDNESS_PATH_GLOBS)}) can squash-auto-merge with zero reviews"
+        ]
+    findings: list[str] = []
+    count = params.get("required_approving_review_count") or 0
+    if count < 1:
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply _project/scripts/ruleset_review_enforcement.py:88]: Fixed in #941 (commit b77ee3ba). `review_enforcement_findings` no longer checks `required_approving_review_count` at all — only `require_code_owner_review`, which is independently sufficient (GitHub still requires an owner approval on a CODEOWNERS-matched path regardless of count) and doesn't gate non-soundness develop PRs. Updated the runbook target state, the admin-enforcement TODO, and the pinned unit tests (including a new `test_required_approving_review_count_is_not_checked` asserting `count=0` with `code_owner=True` passes) to match.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: docs/operations/repo-admin-settings.md [hunk 2]
```diff
@@ -98,6 +98,58 @@ gh api repos/joeharris76/BenchBox/rulesets/15611785 --jq '
...
+require_code_owner_review: true
+```
+
+Live state at this writing: `required_approving_review_count: 0`,
+`require_code_owner_review: false` — not yet enforced. Applying it is a deferred
+repo-admin action: the develop-PR `GITHUB_TOKEN` has no `administration` scope, so it
+can neither read nor write the ruleset; only an admin PAT can.
+
+Verify (manual or release-canary; needs a ruleset-read token such as
+`RULESET_DRIFT_TOKEN`):
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply docs/operations/repo-admin-settings.md:126]: Fixed in #941 (commit b77ee3ba). Reworded the runbook: the "Verify" section now says the check is manual-only and explicitly states it is NOT wired into `release-canary.yml` (which only runs `ruleset_drift_check.py`), until it's added there as a follow-up.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by joeharris76** [comment]
@codex review
